### PR TITLE
feat(dirscan): add max searchee age filter

### PR DESCRIPTION
**Pull request by s0up4200**
Adds a new global Dir Scan setting, `maxSearcheeAgeDays`, to skip searchees when all files in that searchee are older than the configured cutoff. The filter is applied immediately after scan discovery so excluded searchees never enter tracked-file progress or search/injection work for that run.

The UI now includes a clear age-filter control with presets and cutoff preview, and the Dir Scan docs were updated to explain behavior, mtime semantics, and practical examples.

## Summary by CodeRabbit

* **New Features**
  * Added age-based searchee filtering in Directory Scan with a toggle, numeric days input, preset buttons, and live cutoff preview; default 0 disables filtering and inputs are validated as non-negative.

* **Documentation**
  * Added detailed guide covering cutoff calculation, timestamp source, disabling behavior, and concrete examples.

* **Tests**
  * Added unit tests for filtering behavior, boundary cases, and settings validation.

### CODE REVIEW

File: web/src/components/cross-seed/DirScanTab.tsx [hunk 1]
```diff
@@ -940,6 +953,67 @@ function SettingsDialog({ open, onOpenChange, settings, instances }: SettingsDia
...
+
+                <p className="text-xs text-muted-foreground">
+                  Uses file modified time (mtime). A searchee is skipped only when all files in it are older than the cutoff.
+                </p>
+                <p className="text-xs text-muted-foreground">
+                  Current cutoff: {ageFilterCutoffPreview}
+                </p>
+              </>
+            )}
+          </div>
```
[the remark this answers is not in the document]
Comment (s0up4200) [reply web/src/components/cross-seed/DirScanTab.tsx:1015]: Missing aria-labels can be considered nitpicks from here on out.
